include/max_element.hpp

```diff
@@ -21,11 +21,59 @@ namespace bit {
 
 
 
-// Status: to do
+// Status: complete
 template <class ForwardIt>
 constexpr bit_iterator<ForwardIt> max_element(bit_iterator<ForwardIt> first,
     bit_iterator<ForwardIt> last) {
-    (last);
+
+    using word_type = typename bit_iterator<ForwardIt>::word_type;
+    constexpr std::size_t num_digits = binary_digits<word_type>::value;
+
+    bit_iterator<ForwardIt> cursor = first;
+
+    // handle misaligned first word. for example ^00100^101
+    if (!_is_aligned_lsb(cursor)) {
+
+        // if 0 > 1 (version with comparators) we'll need to do this differently
+        // maybe use _padded_read?
+        word_type shifted_first_word = _shift_towards_lsb(*(cursor.base()), 
+            cursor.position());
+
+        if (shifted_first_word > 0) {
+            std::size_t first_set_bit_position = _tzcnt(shifted_first_word);
+            return cursor + first_set_bit_position;
+        } 
+
+        cursor = bit_iterator(std::next(cursor.base()));
+    }
+
+    ForwardIt word_cursor = cursor.base();
+
+    while (!(last.base() == word_cursor) || 
```

Comment:
So here, we are doing 2 conditional statements for each iteration of the for loop:
`!(last.base() == word_cursor)` and `_is_one_past_last_bit(last, word_cursor)`

I think it can be done in 1 conditional statement:

`std::next(word_cursor, _is_aligned_lsb(last)) != last.base()` assuming that the function result is recognized as a constant by the compiler. The statement serves the same purpose, go until there are no more full words.